saltgui/static/scripts/routes/keys.js

```diff
@@ -0,0 +1,155 @@
+
+    list = this.getPageElement().querySelector('#keys_unaccepted');
+    hostnames = keys.minions_pre.sort();
+    for(i = 0; i < hostnames.length; i++) {
+        this._addPreMinion(list, hostnames[i]);
+    }
+    if(hostnames.length == 0)
+        this._addNone(list);
+
+    list = this.getPageElement().querySelector('#keys_rejected');
+    hostnames = keys.minions_rejected.sort();
+    for(i = 0; i < hostnames.length; i++) {
+        this._addRejectedMinion(list, hostnames[i]);
+    }
+    if(hostnames.length == 0)
+        this._addNone(list);
+
+    this.keysLoaded = true;
+    if(this.keysLoaded && this.jobsLoaded) this.resolvePromise();
+  }
+
+  _addMenuItemAccept(menu, hostname) {
+    menu.addMenuItem("Accept&nbsp;key...", function(evt) {
+      this._runAcceptKey(evt, hostname);
+    }.bind(this));
+  }
+
+  _addMenuItemDelete(menu, hostname) {
+    menu.addMenuItem("Delete&nbsp;key...", function(evt) {
+      this._runDeleteKey(evt, hostname);
+    }.bind(this));
+  }
+
+  _addMenuItemReject(menu, hostname) {
+    menu.addMenuItem("Reject&nbsp;key...", function(evt) {
+      this._runRejectKey(evt, hostname);
+    }.bind(this));
+  }
+
+  _updateOfflineMinion(container, hostname) {
+    super._updateOfflineMinion(container, hostname);
+
+    var element = document.getElementById(hostname);
+    var menu = new DropDownMenu(element);
+    this._addMenuItemReject(menu, hostname);
+    this._addMenuItemDelete(menu, hostname);
+  }
+
+  _updateMinion(container, minion) {
+    super._updateMinion(container, minion);
+
+    var element = document.getElementById(minion.hostname);
+    var menu = new DropDownMenu(element);
+    this._addMenuItemReject(menu, minion.hostname);
+    this._addMenuItemDelete(menu, minion.hostname);
+  }
+
+  _addRejectedMinion(container, hostname) {
+    var element = document.createElement('li');
+
+    element.appendChild(Route._createDiv("hostname", hostname));
+
+    var rejected = Route._createDiv("rejected", "rejected");
+    rejected.id = "status";
+    element.appendChild(rejected);
+
+    var menu = new DropDownMenu(element);
+    this._addMenuItemDelete(menu, hostname);
+    this._addMenuItemAccept(menu, hostname);
+
+    container.appendChild(element);
+  }
+
+  _addDeniedMinion(container, hostname) {
+    var element = document.createElement('li');
+
+    element.appendChild(Route._createDiv("hostname", hostname));
+
+    var denied = Route._createDiv("denied", "denied");
+    denied.id = "status";
+    element.appendChild(denied);
+
+    var menu = new DropDownMenu(element);
+    this._addMenuItemAccept(menu, hostname);
+    this._addMenuItemReject(menu, hostname);
+    this._addMenuItemDelete(menu, hostname);
+
+    container.appendChild(element);
+  }
+
+  _addPreMinion(container, hostname) {
+    var element = document.createElement('li');
+
+    element.appendChild(Route._createDiv("hostname", hostname));
+
+    var pre = Route._createDiv("unaccepted", "unaccepted");
+    pre.id = "status";
```

Comment:
Adding multiple elements with the same id is not allowed in the dom. So you should add a class here. See also my general comments about this pull request